tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -2467,3 +2476,249 @@ def test_6_barbican(self):
+                    "Number of snapshots created not equal to retention_policy_value")
+
+            # Check first snapshot is deleted or not after retention value
+            # exceed
+            deleted_snapshot_id = snapshots_list[0]
+            LOG.debug("snapshot id of first snapshot is : " +
+                      str(deleted_snapshot_id))
+            if deleted_snapshot_id in snapshot_list_of_workload:
+                reporting.add_test_step(
+                    "Verify first snapshot deleted after retention value exceeds",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "first snapshot not deleted after retention value exceeds")
+            else:
+                reporting.add_test_step(
+                    "Verify first snapshot deleted after retention value exceeds",
+                    tvaultconf.PASS)
+                LOG.debug("first snapshot deleted after retention value exceeds")
+
+            # Check first snapshot is deleted from backup target when retention
+            # value exceed
+            mount_path = self.get_mountpoint_path(
+                ipaddress=tvaultconf.tvault_ip[0],
+                username=tvaultconf.tvault_username,
+                password=tvaultconf.tvault_password)
+            LOG.debug("Backup target mount_path is : " + mount_path)
+            is_snapshot_exist = self.check_snapshot_exist_on_backend(
+                tvaultconf.tvault_ip[0],
+                tvaultconf.tvault_username,
+                tvaultconf.tvault_password,
+                mount_path,
+                self.workload_id,
+                deleted_snapshot_id)
+            LOG.debug("Snapshot does not exist : %s" % is_snapshot_exist)
+            if not is_snapshot_exist:
+                LOG.debug("First snapshot is deleted from backup target")
+                reporting.add_test_step(
+                    "First snapshot deleted from backup target",
+                    tvaultconf.PASS)
+            else:
+                reporting.add_test_step(
+                    "First snapshot deleted from backup target",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "First snapshot is not deleted from backup target media")
+
+
+        except Exception as e:
+            LOG.error("Exception: " + str(e))
+            reporting.set_test_script_status(tvaultconf.FAIL)
+        finally:
+            reporting.test_case_to_write()
+            # Cleanup
+            # Delete snapshot
+            snapshot_list_of_workload = self.getSnapshotList(self.workload_id)
+            for i in range(0, len(snapshot_list_of_workload)):
+                self.snapshot_delete(
+                    self.workload_id, snapshot_list_of_workload[i])
+
+            # Delete workload
+            self.workload_delete(self.workload_id)
+            time.sleep(15)
+
+            # Delete policy
+            self.workload_policy_delete(policy_id)
```

Comment:
This cleanup can be handled in policy creation step itself. Any specific reason for explicitly adding the delete call?